### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by doxliydriw**
Profile page which in this PR should:
- show present user's information: avatar, name, email, about.
- toggle the form from VIEW to EDIT
- save updated users information: name, email, about;
- after successful information update return to profile page with view mode form.

(presently users information is set in form from Redux store and then request for update is sent to server)

My present problem is that after successful information update the form is re-rendered empty, without users data.
Check Profile.jsx file line 59: onSubmit

### CODE REVIEW

File: src/components/InputField/InputFieldStyled.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import styled from 'styled-components';
 
-const getNotificationColor = ({ $isErrorColor, $isSuccessColor }) => {
+const getNotificationColor = ({ $isErrorColor, $isSuccessColor }) =>
+{
   if ($isErrorColor) return 'var(--color-error)';
   if ($isSuccessColor) return 'var(--color-success)';
   return '#79747E';
```
Comment (ReginaVarybrus) [inline src/components/InputField/InputFieldStyled.js]: I propose to add colors such as '#79747E' to the GlobalStyles

File: src/components/InputField/InputFieldStyled.js [hunk 2]
```diff
@@ -17,7 +18,16 @@ export const InputFieldStyled = styled.div`
...
 
 export const StyledLabel = styled.label`
-  color: ${getNotificationColor};
+  position: absolute;
+  left: 1px;
+  transform: translateY(-50%);
+  background: var(--white-color);
+  padding: 0px 4px 0px 4px;
+  margin: 0 12px;
+  color: #49454F;
```
Comment (ReginaVarybrus) [inline src/components/InputField/InputFieldStyled.js]: Same lets add it to the GlobalStyles

File: src/components/InputField/InputFieldStyled.js [hunk 3]
```diff
@@ -27,10 +37,10 @@ export const StyledInput = styled.input`
   height: 56px;
   border-radius: 8px;
   border: 1px solid;
-  border-color: ${getNotificationColor};
+  border-color: ${({ disabled }) => (disabled ? 'transparent' : getNotificationColor)};
 
   &:hover {
-    border: 1px solid #111;
+    border: 1px solid ${({ disabled }) => (disabled ? 'transparent' : '#111')};;
```
Comment (ReginaVarybrus) [inline src/components/InputField/InputFieldStyled.js]: Same lets add it to the GlobalStyles

File: src/components/Profile/Profile.jsx [hunk 4]
```diff
@@ -1,219 +1,148 @@
...
-              {/* <VectorPng>
-                  <use href={Icon + '#icon-chevron-right'}></use>
-                </VectorPng>
+  const formik = useFormik({
+    initialValues: user,
+    validationSchema: schema,
+    validateOnChange: false,
+    enableReinitialize: true,
+    onSubmit: async (values, { resetForm }) => {
+      // console.log('SUBMIT', values);
```
Comment (ReginaVarybrus) [inline src/components/Profile/Profile.jsx]: Remove all console.log before the push your request

File: src/components/Profile/ProfileStyled.js [hunk 5]
```diff
@@ -1,347 +1,65 @@
...
-    cursor: pointer;
-    background-color: #2b78ef;
-  }
-  &:active {
-    filter: blur(0.1rem);
-  }
+export const EditButton = styled.button`
+width: 18px;
+height: 18px;
+background-image: url(${props => props.$icon === 'edit' ? PencilIcon : CloseIcon});
```
Comment (ReginaVarybrus) [inline src/components/Profile/ProfileStyled.js]: Lets try to do is this way `background-img: url(${({ $icon }) => ($icon ? PencilIcon : CloseIcon)}); `

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/components/Profile/Profile.jsx
Comment (ReginaVarybrus) [inline src/components/Profile/Profile.jsx:1]: I propose to moved this file to the Account Route.

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/components/Buttons/BasicButton/BasicButtonStyled.js [hunk 6]
```diff
@@ -11,7 +11,6 @@ export const BasicButtonLayout = styled(Button)`
     line-height: 19.6px;
     text-align: center;
     padding: 12px 24px;
-    width: Hug (92px)px;
     border-radius: 4px;
     border: 1px 0px 0px 0px;
```
Comment (ReginaVarybrus) [inline src/components/Buttons/BasicButton/BasicButtonStyled.js]: Maybe here is dhould be just border-rop: 1px; ?

File: src/routes/AccountRoute/AccountRouteStyled.js [hunk 7]
```diff
@@ -30,15 +33,23 @@ line-height: 28.8px;
...
 min-height: 320px;
 box-sizing: border-box;
 
 @media (max-width: 767px) {
-    flex-direction: column;
+grid-template-columns: 1fr auto;
+    grid-template-areas:
+      "avatar edit" 
+      "form form"
+      "logout logout";
```
Comment (ReginaVarybrus) [inline src/routes/AccountRoute/AccountRouteStyled.js]: wow didn`t see same before.

File: src/routes/AccountRoute/AccountRouteStyled.js [hunk 8]
```diff
@@ -67,12 +78,17 @@ height: 64px;
...
 `;
 
-export const ChangeAvatar = styled.a`
+export const ChangeAvatar = styled(BasicButton)`
 padding-top: 8px;
 color: var(--color-brand-blue);
+width: 120px;
+
+@media (max-width: 767px) {
+}
```
Comment (ReginaVarybrus) [inline src/routes/AccountRoute/AccountRouteStyled.js:89]: If you don't need this mediaQuery let's remove it

File: src/routes/AccountRoute/AccountRouteStyled.js [hunk 9]
```diff
@@ -100,6 +117,7 @@ border: 1px solid;
 border-color: transparent;
 `
 export const EditButton = styled.button`
+grid-area: edit;
 width: 18px;
 height: 18px;
 background-image: url(${props => props.$icon === 'edit' ? PencilIcon : CloseIcon});
```
Comment (ReginaVarybrus) [inline src/routes/AccountRoute/AccountRouteStyled.js:125]: I commited earlier to try do it another way. Did you try?

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/routes/AccountRoute/AccountRoute.jsx [hunk 10]
```diff
@@ -1,5 +1,158 @@
...
+          // console.log('Update Successful:', resultAction.payload);
+          navigate(routesPath.ACCOUNT);
+          resetForm();
+          // console.log('we are here wtf?');
+          setEditMode(false);
+        } else {
+          // console.error('Update Failed:', resultAction.error);
+        }
+      } catch (error) {
+        // console.error('Something went wrong:', error);
```
Comment (ReginaVarybrus) [inline src/routes/AccountRoute/AccountRoute.jsx]: Do you plan to remove commentes lines?

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/routes/AccountRoute/AccountRoute.jsx (hunk 10 above)
Comment (doxliydriw) [reply src/routes/AccountRoute/AccountRoute.jsx]: Removed comment lines.

File: src/components/InputField/InputFieldStyled.js (hunk 1 above)
Comment (doxliydriw) [reply src/components/InputField/InputFieldStyled.js]: Changed all text colors to var(--color-grey-6)

File: src/components/InputField/InputFieldStyled.js (hunk 2 above)
Comment (doxliydriw) [reply src/components/InputField/InputFieldStyled.js]: Changed all text colors to var(--color-grey-6)

File: src/components/InputField/InputFieldStyled.js (hunk 3 above)
Comment (doxliydriw) [reply src/components/InputField/InputFieldStyled.js]: Changed to 'var(--color-dark)'.

File: src/components/Profile/Profile.jsx (hunk 4 above)
Comment (doxliydriw) [reply src/components/Profile/Profile.jsx]: done

File: src/components/Profile/Profile.jsx
Comment (doxliydriw) [reply src/components/Profile/Profile.jsx:1]: done

File: src/routes/AccountRoute/AccountRouteStyled.js (hunk 7 above)
Comment (doxliydriw) [reply src/routes/AccountRoute/AccountRouteStyled.js]: lazy grid area marking )

File: src/routes/AccountRoute/AccountRouteStyled.js (hunk 8 above)
Comment (doxliydriw) [reply src/routes/AccountRoute/AccountRouteStyled.js:89]: need i.

File: src/components/Profile/ProfileStyled.js (hunk 5 above)
Comment (doxliydriw) [reply src/components/Profile/ProfileStyled.js]: I am providing conditionally variable $icon into component:
<EditButton
          onClick={editMode ? cancelEdit : toggleEditMode}
          $icon={editMode ? 'close' : 'edit'}
        />
        
and compare check if it is 'edit' or not to set appropriate background-image image.

File: src/components/Buttons/BasicButton/BasicButtonStyled.js (hunk 6 above)
Comment (doxliydriw) [reply src/components/Buttons/BasicButton/BasicButtonStyled.js]: this was changed.

**Pull request merged**